test/f90_correct/src/common_01.f90

```diff
@@ -0,0 +1,37 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! Check if two common block variables are equal when turn on optimization
+
+module mod1
+  integer :: mi1
+  common /com1/ mi1
+contains
+  subroutine newmi1()
+    mi1 = 30
+  end subroutine newmi1
+end module mod1
+
+program test
+  use mod1
+  integer, parameter :: n = 6
+  integer :: res(n), rres(n)
+  integer :: pi1
+  common /com1/ pi1
+
+  res = 0
+  rres= 1
+  pi1 = 10
+  mi1 = 20
+  if (pi1 == mi1) res(1) = 1
```

Comment:
Not a big deal, but if you use the `check_mod` module, you could declare the `reslt` and `expct` arrays as `logical(n)` and these `if` statements could be reduced to:
```fortran
reslt(1:3) = (/pi1 == mi1, pi1 == 20, mi1 == 20/)
```

Reply:
I've modified the code as suggested. Thanks!

Reply:
I've modified the code as suggested. Thanks!